internal/db/sqlite/folderdb_update.go

```diff
@@ -46,10 +46,28 @@ func (s *folderDB) Update(device protocol.DeviceID, fs []protocol.FileInfo) erro
 	defer tx.Rollback() //nolint:errcheck
 	txp := &txPreparedStmts{Tx: tx}
 
+	//nolint:sqlclosecheck
+	insertNameStmt, err := txp.Preparex(`
+		INSERT OR IGNORE INTO file_names (name)
+		VALUES (?)
+	`)
+	if err != nil {
+		return wrap(err, "prepare insert name")
+	}
+
+	//nolint:sqlclosecheck
+	insertVersionStmt, err := txp.Preparex(`
+		INSERT OR IGNORE INTO file_versions (version)
+		VALUES (?)
```

Comment:
Could these two new INSERTs return the generated index integer, to avoid another string lookup in the next query?

Reply:
Is it better?

Reply:
Yes, it will always perform better, as IGNORE discards the conflict results. CONFLICT does the name lookup for us "for free".

We can then skip the lookups:
```
INSERT OR REPLACE INTO files (device_idx, remote_sequence, type, modified, size, deleted, local_flags, blocklist_hash, name_idx, version_idx)
		VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
```